dbms/src/Storages/StorageDisaggregated.cpp

```diff
@@ -0,0 +1,355 @@
+                                          sender_target_task_start_ts,
+                                          sender_target_task_task_id));
+
+    tipb::ExchangeSender * sender = executor->mutable_exchange_sender();
+    sender->set_tp(tipb::ExchangeType::PassThrough);
+    sender->add_encoded_task_meta(sender_target_task_meta.SerializeAsString());
+    auto * child = sender->mutable_child();
+    child->CopyFrom(buildTableScanTiPB());
+    for (const auto & column_info : column_infos)
+    {
+        auto * field_type = sender->add_all_field_types();
+        auto tidb_column_info = TiDB::toTiDBColumnInfo(column_info);
+        *field_type = columnInfoToFieldType(tidb_column_info);
+    }
+    // Ignore sender.PartitionKeys and sender.Types because it's a PassThrough sender.
+
+    dispatch_req->set_encoded_plan(sender_dag_req.SerializeAsString());
+    return StorageDisaggregated::RequestAndRegionIDs{dispatch_req, region_ids, batch_cop_task.store_id};
+}
+
+tipb::Executor StorageDisaggregated::buildTableScanTiPB()
+{
+    // TODO: For now, to avoid versions of tiflash_compute nodes and tiflash_storage being different,
+    // disable filter push down to avoid unsupported expression in tiflash_storage.
+    // Uncomment this when we are sure versions are same.
+    // executor = push_down_filter.constructSelectionForRemoteRead(dag_req.mutable_root_executor());
+
+    tipb::Executor ts_exec;
+    ts_exec.set_tp(tipb::ExecType::TypeTableScan);
+    ts_exec.set_executor_id(table_scan.getTableScanExecutorID());
+
+    // In disaggregated mode, use DAGRequest sent from TiDB directly, so no need to rely on SchemaSyncer.
+    if (table_scan.isPartitionTableScan())
+    {
+        ts_exec.set_tp(tipb::ExecType::TypePartitionTableScan);
+        auto * mutable_partition_table_scan = ts_exec.mutable_partition_table_scan();
+        *mutable_partition_table_scan = table_scan.getTableScanPB()->partition_table_scan();
+    }
+    else
+    {
+        ts_exec.set_tp(tipb::ExecType::TypeTableScan);
+        auto * mutable_table_scan = ts_exec.mutable_tbl_scan();
+        *mutable_table_scan = table_scan.getTableScanPB()->tbl_scan();
+    }
+    return ts_exec;
+}
+
+std::vector<StorageDisaggregated::RequestAndRegionIDs>
+StorageDisaggregated::buildAndDispatchMPPTaskRequests(const std::vector<RemoteTableRange> & remote_table_ranges)
+{
+    auto batch_cop_tasks = buildBatchCopTasks(remote_table_ranges);
+    std::vector<RequestAndRegionIDs> dispatch_reqs;
+    dispatch_reqs.reserve(batch_cop_tasks.size());
+    for (const auto & batch_cop_task : batch_cop_tasks)
+        dispatch_reqs.emplace_back(buildDispatchMPPTaskRequest(batch_cop_task));
+
+    std::shared_ptr<ThreadManager> thread_manager = newThreadManager();
+    for (const RequestAndRegionIDs & dispatch_req : dispatch_reqs)
+    {
+        LOG_DEBUG(log, "tiflash_compute node start to send MPPTask({})", std::get<0>(dispatch_req)->DebugString());
```

Comment:
Maybe we should dispatch the task after current mpp task is scheduled?

Reply:
After discuss, for now, we still put dispatch in compilation phase.